src/27-ipfs-at-the-vatican/index.md

```diff
@@ -0,0 +1,158 @@
+
+
+If you look at the URIs in all of these examples, you'll see they all rely on the `https://ipfs.io` domain. These are _gateway links_, which make IPFS content backwards-compatible with the classical HTTP web. Every IPFS node is able to present itself as an HTTP server, acting as a _gateway_ between the HTTP web and the distributed peer-to-peer web. When one of these gateways receives a request, it uses the IPFS protocol to retrieve the corresponding content from the peer-to-peer web and then returns it over HTTP.
+
+If you use one of these gateway links in a web browser that doesn't speak IPFS, the content will still load because you're falling relying on the gateway to translate the requests for you. Over time, as browsers add support for IPFS (starting with the IPFS web extensions you can install in your browser) the browser will recognize those gateway requests, skip the HTTP connection, and instead resolve the content via IPFS.
+
+In the meantime, to confirm that you're actually retrieving content that any IPFS node could resolve, you can replace the ipfs.io portion of any url with the hostname of any IPFS gateway. There is a [thread in the IPFS discussion forum](https://discuss.ipfs.io/t/curated-list-of-ipfs-gateways/620) where people are gathering a list of gateways you could test out.
+
+## The Second Demo: Serverless Collaborative Annotations on IPFS
+
+Our second demo is more involved and it requires more explanation. The demo focuses on allowing people to annotate images and sync those annotations across the peer-to-peer network without relying on an annotation server. To support this real-time collaborative sharing of data we use [CRDTs](https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type) and [IPFS pubsub](../25-pubsub/).
+
+One obstacle to enabling annotations of IIIF content is the fact that nobody wants to run an annotation server. Institutions want to let their own patrons and curators create annotations. Though they are also interested in seeing the annotations other people create, they don't want to get stuck storing the whole world's annotations indefinitely. That would be like your local public library committing to store all the notes anyone ever made about the books in their collection rather than letting everyone take their own notes home with them and store those notes as they see fit.
+
+Peer to peer tools are the right tech to apply here. With the demo we set up, each node on the network holds its own copy of the annotations. The user of that node decides which streams of annotations they will publish to and subscribe from. Groups of people can share their annotations as they see fit. Third parties like the library can follow or ignore public streams, choosing which content to keep a copy of.
```

Comment:
Maybe worth noting here that CRDTs allow us to stop listening and then catch up later once the first new update comes in. What I mean is: it's resistant to message loss because each message points to some previous history (depending on the sender's view of the state). @pgte is that accurate?